tests: kernel: timer behavior lower SYS_CLOCK_TICKS_PER_SEC

FRASTM opened the pull request:
Reduce the CONFIG_SYS_CLOCK_TICKS_PER_SEC below 10000 when executing the tests/kernel/timer/timer_behavior on stm32l073 and stm32wba55 nucleo boards.
Both have a low sysclock freq compared to the ticks per sec

Other stm32 target boards with low sysclock cannot pass the testcase due to flash/ram capacity

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/75673

andyross reviewed (commented):
Nitpick: isn't the root cause then that the platform as a whole has a tick rate that is too high?  The tick rate is a promise: the system claims that it can handle general timing with 100us precision.  If that's not working due to hardware limitations (or driver bugs that cause latency glitches, etc...) then the correct fix should be to adjust what the platform is promising and not just patch the test that's failing.

Not a -1 if there are special circumstances to this test or whatever, but we should know and document them if there are.

npitre commented:
Same sentiment here. Isn't the default tick rate simply too high for the 
platform?

FRASTM committed:
soc: stm32 devices have lower tick with lower sysclock

For stm32 platforms where the sysclock is less or equal to
32MHz, the Ticks per second is reduced to 8000 (instead of
10000).

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
> Same sentiment here. Isn't the default tick rate simply too high for the platform?

Yes, that's the case. and for other with similar sysclock. 
--> I 'll put a Kconfig condition on TICKS_PER_SEC for any stm32

FRASTM commented:
Change the CONFIG_SYS_CLOCK_TICKS_PER_SEC for any stm32 platform with a CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC <= MHZ(32)
